scripts/ci/build.sh

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+set -ex
+
+ROOTDIR="$(dirname "$0")/../.."
+cd "${ROOTDIR}"
+
+# shellcheck disable=SC2166
+if [ "$CIRCLE_BRANCH" = release -o -n "$CIRCLE_TAG" -o -n "$FORCE_RELEASE" ]; then echo -n >prerelease.txt; else date -u +"nightly.%Y.%-m.%-d" >prerelease.txt; fi
+if [ -n "$CIRCLE_SHA1" ]
+then
+  echo -n "$CIRCLE_SHA1" >commit_hash.txt
+fi
```

Comment:
This would be more readable on multiple lines. The `else` is really easy to miss here. How about splitting it up a bit?
``` suggestion
[ "$CIRCLE_BRANCH" = release -o -n "$CIRCLE_TAG" -o -n "$FORCE_RELEASE" ] && echo -n >prerelease.txt
[ "$CIRCLE_BRANCH" != release -a -z "$CIRCLE_TAG" -a -z "$FORCE_RELEASE" ] && date -u +"nightly.%Y.%-m.%-d" >prerelease.txt
[ -n "$CIRCLE_SHA1" ] && echo -n "$CIRCLE_SHA1" >commit_hash.txt
```

By the way, why not use the `[[]]` conditions and long options (`date --utc` rather than `date -u`)? This is supposed to run inside a docker images in CI so I don't think we have to worry about portability all that much here.

EDIT: Oh, I see that it's how it originally was in the CircleCI config. In that case I guess leaving it as is is fine. But I'd still be tempted to clean it up a bit :)

Reply:
We can do it in a separate step, I'd keep this file as close as possible to the original CircleCI config here for now - a lot of that can indeed use some cleanup, but changing it here requires waiting an hour to confirm the builds still work, whereas changing it later is just a regular PR :-).

Reply:
Fair enough.